ext/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+ # DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ # ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ # (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ # LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ # ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ #
+ #################################################################################
+
+# Dependencies between libraries are solved using the given order
+# Note: We prefer building dependencies our self to using find_library,
+# since it does not necessarily give you the dll for packaging.
+# It also ensures that we are in control of the used version
+# and is built using the same, Inviwo, settings.
+
+ivw_append_install_list(DIRECTORY)
+
+# GTest
+option(IVW_USE_EXTERNAL_GTEST "GTest is provided externaly" OFF)
+ # don't install, used in build only
+if(NOT IVW_USE_EXTERNAL_GTEST AND (IVW_TEST_UNIT_TESTS OR IVW_TEST_INTEGRATION_TESTS))
+	add_subdirectory(gtest)
+	ivw_register_license_file(NAME "Google C++ Testing Framework" TARGET gtest MODULE Core EXT
+        URL "https://github.com/google/googletest"
+        TYPE "BSD 3-clause"
+        FILES ${IVW_EXTENSIONS_DIR}/gtest/gtest/LICENSE
+    )
+else()
+	find_package(GTest CONFIG REQUIRED)
+endif()
+
+# Img foramts
+option(IVW_USE_EXTERNAL_IMG "Link with external image libraries (zlib, png, jpeg, tiff) instead of building them." OFF)
+if (NOT IVW_USE_EXTERNAL_IMG)
+ 	add_subdirectory(zlib-1.2.11) # Defines target ZLIB::ZLIB
+ 	add_subdirectory(libpng)      # Defines target png
+ 	add_subdirectory(libjpeg)     # Defines target JPEG::JPEG
+ 	add_subdirectory(tiff)        # Defines target TIFF::TIFF
+
+	ivw_register_license_file(NAME "zlib" TARGET ZLIB::ZLIB MODULE Core EXT
+	    URL "http://zlib.net/" TYPE "The zlib/libpng License"
+	    FILES ${IVW_EXTENSIONS_DIR}/zlib-1.2.11/LICENSE
+	)
+	ivw_register_license_file(NAME libjpeg-turbo TARGET JPEG::JPEG MODULE Core EXT
+	    TYPE "BSD-style, acknowledgement: This software is based in part on the work of the Independent JPEG Group"
+	    URL www.ijg.org FILES ${IVW_EXTENSIONS_DIR}/libjpeg/README
+	)
+	ivw_register_license_file(NAME tiff TARGET TIFF::TIFF MODULE Core EXT
+	    TYPE "The MIT License" URL http://www.libtiff.org/
+	    FILES ${IVW_EXTENSIONS_DIR}/tiff/COPYRIGHT
+	)
+	ivw_register_license_file(NAME libpng TARGET png MODULE Core EXT 
+		TYPE "libpng (BSD-style)" URL http://www.libpng.org/pub/png/libpng.html
+	    FILES ${IVW_EXTENSIONS_DIR}/libpng/LICENSE.txt
+	)
+else()
+	find_package(ZLIB REQUIRED)
+	find_package(JPEG REQUIRED)
+	find_package(TIFF REQUIRED)
+	find_package(PNG  REQUIRED)
+	ivw_vcpkg_install(zlib          MODULE Core EXT)
```

Comment:
could potentially fail

Reply:
should fail in a nice way, i.e. do nothing